### PR TITLE
applies session-store of groups as whitelist to group list ui

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/search/group/DefaultGroupService.java [hunk 1]
```diff
@@ -24,7 +25,9 @@ public DefaultGroupService(@NotNull final GroupRepository groupRepository) {
         if (!user.getPermissionsById().containsKey("GROUP_PII_READ")) {
             return ImmutableSet.of();
         }
-        return newHashSet(groupRepository.findAllForUsername(user.getUsername()));
+        return groupRepository.findAllForUsername(user.getUsername()).stream()
```
Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/search/group/DefaultGroupService.java]: I think classic collection functions like retainAll would be more efficient (avoid creating extra containers) but this'll work ...

Comment (esotrope) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/search/group/DefaultGroupService.java]: I think retain all relies on ```Object.equals``` though which i don't think we want here.

Comment (bgsavage) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/search/group/DefaultGroupService.java]: I thought you would want to pass the group ids to the query like we do everywhere else.

Comment (esotrope) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/search/group/DefaultGroupService.java]: True, that would be slightly cleaner - thanks

**Pull request merged**